Add julia to deployment

pramodk opened the pull request:
Fixes #200
Closes #199

pramodk committed:
Add julia to deployment

pramodk committed:
Revert "Add julia to deployment"

This reverts commit cbf0654264976a0bf8f12600bc3d6a18742d6a92.

pramodk committed:
add julia in serial libraries for python dependencies

pramodk committed:
julia with python2

pramodk committed:
julia with out cxx

pramodk commented:
@matz-e : could you take a look here when you have some time?

matz-e commented:
@pramodk that's a very non-descript failure to me.

Retest this please

pramodk commented:
@matz-e : I think we have to build this locally on BB5 and track the issue

matz-e commented:
Did we have a JIRA ticket for this?

matz-e commented:
Investigation so far from [discourse](https://discourse.julialang.org/t/source-build-julia-0-6-0-pre-alpha-branch-fails-with-error-in-inference-ji/10689/5)

> Set MARCH=haswell in Make.user solved the issue.
> There is another post that states the same issue, I am pasting the answer here:
> “So this happens to be an issue with LLVM on Skylake CPUs. It should be fixed on master but meanwhile you can work around the issue by setting MARCH=haswell when building Julia.”

matz-e commented:
@pramodk why do we use 0.6.2? Shouldn't we use 1.0 or 1.1? This version here is EOL: [see the announcement](https://discourse.julialang.org/t/julia-0-6-and-0-7-are-now-officially-unmaintained/20449)

pramodk commented:
* yes, there is a ticket related to Julia from Polina (see Julia integration with NEURON)
* if newer version builds, we should use it.

> from my tiny mobile screen.

matz-e committed:
Bump version.

Avoids compilation issue on Skylakes and also moves us to a *supported*
version of Julia.

matz-e committed:
Fix Julia breakage.

matz-e commented:
Please retest.

Wondering if that's a transient failure… Github not being reachable.

matz-e committed:
Fiddle around with Julia.

matz-e committed:
julia: move to 1.2.0 + trunk, use external LLVM

Build with the proper LLVM version, use not rc1 but newer revision to
have working build system customizations (without patching build system
files).

Error message «du jour»:

    ┌ Warning: No working GUI backend found for matplotlib
    └ @ PyPlot ~/.cache/spack/linux-rhel7-x86_64/gcc-6.4.0/julia-1.2.0-rc1-snapshot0-d2gfb5/var/julia/pkg/packages/PyPlot/cdCMF/
    src/init.jl:155
    ERROR: InitError: PyError (PyImport_ImportModule

    The Python package matplotlib.pyplot could not be found by pyimport. Usually this means
    that you did not install matplotlib.pyplot in the Python version being used by PyCall.

matz-e committed:
Working version of julia!

matz-e committed:
Hook julia up better.

matz-e committed:
Fix llvm variants.

matz-e committed:
Gunk things up with more module glue.

Add --latest to module's find command and respect it with spack load.

matz-e commented:
Please retest

matz-e committed:
D-bug.

matz-e committed:
Less debug

matz-e committed:
D+bug

matz-e committed:
E+bug

matz-e committed:
F-bug

matz-e committed:
One step back.

matz-e committed:
A-nother attempt.

matz-e committed:
Spurious )

matz-e committed:
Better load/unload of modules.

matz-e committed:
Keep intel from showing up places.

matz-e committed:
Actually build julia.

matz-e committed:
Force no binary builder.

matz-e committed:
Use rc1.

matz-e committed:
Set correct builder values.

matz-e commented:
Retest this please

matz-e committed:
Unforce more binary builder stuff.

matz-e committed:
Move to makefile build system.

A failing install phase would leave too much wreakage behing.

matz-e commented:
Retest this please

matz-e committed:
Polish julia setup.

Polina's test script works now, without installing additional
dependencies. Julia's package depots are neatly separated.

matz-e committed:
Fiddle with LLVM dependency.

matz-e committed:
Fiddle with LLVM dependency.

matz-e commented:
Retest this please

matz-e commented:
Retest this please

matz-e committed:
deployment: brute-force remove escape sequences.

Prevent XML parsing failures by removing output from lazy software
packages that don't use isatty() and pollute their output with escape
sequences.

matz-e commented:
Retest this please

matz-e reviewed (approved):
I approve!

@pramodk any last words before I merge this? NGV-103 tests have been successful.

matz-e committed:
Packages become user responsibility.

matz-e commented on deploy/configs/serial-libraries/modules.yaml:
```diff
@@ -31,6 +31,7 @@ modules:
       - gmsh
       - hdf5~mpi
       - highfive~mpi
+      - julia-dev
```
```suggestion
```

matz-e committed:
Update deploy/configs/serial-libraries/modules.yaml

matz-e committed:
Fixes.

matz-e committed:
Enable pure julia module.

pramodk commented:
good to merge! I missed your last comment.
